src/Tribe/PUE/Checker.php

```diff
@@ -518,98 +573,89 @@ private function get_api_message( $info ) {
-				$license_tab_link = sprintf( '<a href="' . $license_tab . '">%s</a>', esc_html__( 'Add your license key', 'tribe-common' ) );
-				$tec_link = '<a href="https://theeventscalendar.com" target="_blank">' . esc_html__( 'theeventscalendar.com', 'tribe-common' ) . '<span class="screen-reader-text">' .  esc_html__( 'opens in a new window', 'tribe-common' ) . '</span></a>';
-				$link   = '<a href="http://m.tri.be/195d" target="_blank">' . esc_html__( 'license keys', 'tribe-common' ) . '<span class="screen-reader-text">' .  esc_html__( 'opens in a new window', 'tribe-common' ) . '</span></a>';
-				$html[] = '<p>' . sprintf( __( '%s so that you can always have access to the latest versions including bug fixes, security updates, and new features.', 'tribe-common' ), $license_tab_link ) . '</p>';
-				$html[] = '<p>' . sprintf( __( 'You can find your %1$s in your account on %2$s.', 'tribe-common' ), $link, $tec_link ) . '</p>';
-			}
-			return Tribe__Admin__Notices::instance()->render( 'pue-validation', implode( "\r\n", $html ) );
-		}
-
-		public function get_license_expired_message() {
-			$expired_message = '<a href="http://m.tri.be/195y" target="_blank" class="button button-primary">' .
-			__( 'Renew Your License Now', 'tribe-common' ) .
-			'<span class="screen-reader-text">' .
-			__( ' (opens in a new window)', 'tribe-common' ) .
-			'</span></a>';
+			$message = sprintf(
+				esc_html__( 'There is an update for %s. %sRenew your license%s to get access to bug fixes, security updates, and new features.', 'tribe-common' ),
+				$this->get_plugin_name(),
+				'<a href="https://theeventscalendar.com/license-keys/">',
+				'</a>'
+			);
 
-			return $expired_message;
+			return $message;
 		}
 
 		/**
-		 * Displays a PUE message on the page if it is relevant
+		 * Returns plugin/license key data based on the provided query arguments.
+		 *
+		 * Calling this method will also take care of setting up admin notices for any
+		 * keys that are invalid or have expired, etc.
+		 *
+		 * @see Tribe__PUE__Checker::request_info()
+		 *
+		 * @param $query_args
+		 *
+		 * @return Tribe__PUE__Plugin_Info|null
 		 */
-		public function maybe_display_json_error_on_plugins_page( $page ) {
-			if ( 'plugins.php' !== $page ) {
-				return;
-			}
-
-			$state = $this->get_option( $this->pue_option_name, false, false );
+		public function license_key_status( $query_args ) {
+			$pue_notices = Tribe__Main::instance()->pue_notices();
+			$plugin_info = $this->request_info( $query_args );
+			$plugin_name = empty( $this->plugin_name ) ? $this->get_plugin_name() : $this->plugin_name;
 
-			if ( empty( $state->update->license_error ) ) {
-				return;
+			if ( empty( $plugin_name ) ) {
+				return $plugin_info;
 			}
 
-			$this->plugin_notice = array(
-				'slug' => $this->get_slug(),
-				'message' => $state->update->license_error,
-			);
-			add_filter( 'tribe_plugin_notices', array( $this, 'add_notice_to_plugin_notices' ) );
```

Comment:
I think you're right actually and I got trigger happy. Will confirm and probably restore.